pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -184,10 +184,13 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 
 		secret := r.secretForInfinispan(identities, infinispan)
```

Comment:
`secretForInfinispan` calls `SetControllerReference`, I think it's ok to call it also on a user provided secret, but I'm not 100% sure

Reply:
Good point, @rigazilla . But if we will do this, user provided secret will remove if user delete Infinispan resource

Reply:
I've just checked: currently the user secret is deleted with the infinispan resource, due to this call:

https://github.com/dmvolod/infinispan-operator/blob/8c14d3e9631241648b6b785a4d011d5b164a5119/pkg/controller/infinispan/infinispan_controller.go#L1070

we probably need to fix the logic here: we may want to watch for the secret changes but not the final "cleanup"

Reply:
Thanks, we will need to change logic to set reference just for auto-generated secrets